### PR TITLE
Bump node to 16 and add missing dev packages

### PR DESCRIPTION
**Pull request by josemigallas**
To verify this, update your `.tool-versions`, run `asdf install` then re-install dependencies and compile assets in production mode:
```
yarn clean && yarn --production
bundle
NODE_ENV=production RAILS_ENV=production rails assets:precompile
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,59 +29,81 @@
...
     "typescript": "5.0.4",
     "webpack": "4.46.0",
     "webpack-bundle-analyzer": "^4.8.0",
     "webpack-cli": "3.3.12",
-    "webpack-dev-server": "3.11.3",
-    "yaml-loader": "^0.6.0"
+    "webpack-dev-server": "3.11.3"
   },
   "dependencies": {
+    "@babel/core": "7.21.8",
```
Comment (mayorova) [inline package.json:48]: So, we need all these deps in the "dependencies" rather than "devDependencies"?...

Comment (akostadinov) [reply package.json:48]: Ops, I missed this comment. Yes, we need a number of deps to pre-compile assets.

```
$ RAILS_ENV=production NODE_ENV=production SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear
....
EntryModuleNotFoundError: Entry module not found: Error: Can't resolve 'babel-loader' in '/opt/system'
    at /opt/system/node_modules/webpack/lib/Compilation.js:1075:31
    ...
```

Comment (mayorova) [reply package.json:48]: Aha, OK, thanks @akostadinov !

Comment (josemigallas) [reply package.json:48]: Yeah, dev dependencies should be just linters and dev server. Everything required to make the app run (compilation, types, etc) has to be included as a dependency.

Comment (akostadinov) [reply package.json:48]: With node 16. Idk why it worked with node 14.

Comment (mayorova) [reply package.json:48]: yeah, weird, right? :upside_down_face:

**Pull request merged**